optashift-employee-rostering-shared/src/main/java/org/optaplanner/openshift/employeerostering/shared/tenant/TenantConfiguration.java

```diff
@@ -18,55 +18,66 @@
 import com.fasterxml.jackson.annotation.JsonManagedReference;
 import com.fasterxml.jackson.annotation.ObjectIdGenerators;
 import org.optaplanner.core.api.domain.lookup.PlanningId;
+import org.optaplanner.openshift.employeerostering.shared.common.AbstractPersistable;
 
 @Entity
-public class TenantConfiguration {
-
-    @OneToOne(mappedBy = "configuration", fetch = FetchType.EAGER, cascade = CascadeType.ALL, optional = false)
-    @JsonBackReference
-    private Tenant tenant;
-
-    @Id
-    @GeneratedValue(strategy = GenerationType.AUTO)
-    @PlanningId
-    protected Integer id;
-    @Version
-    protected Long version;
-
+public class TenantConfiguration extends AbstractPersistable {
     @NotNull
-    @OneToOne(fetch = FetchType.EAGER, cascade = CascadeType.ALL, optional = false)
-    private TenantConfigurationView view;
+	private Integer undesiredTimeSlotWeight;
+	@NotNull
+	private Integer desiredTimeSlotWeight;
+    @NotNull
+    private Integer templateDuration;
+    @NotNull
+    private DayOfWeek weekStart;
 
     @SuppressWarnings("unused")
     public TenantConfiguration() {
-        view = new TenantConfigurationView(null, 1, DayOfWeek.MONDAY);
+    	templateDuration = 1;
```

Comment:
indentation is 4 spaces, please import this in your IDE: https://github.com/kiegroup/droolsjbpm-build-bootstrap/tree/master/ide-configuration